src/stream/src/executor/temporal_filter.rs

```diff
@@ -0,0 +1,466 @@
+                            (time_millis / 1000) as i64,
+                            (time_millis % 1000 * 1_000_000) as u32,
+                        )),
+                        window_size,
+                    )?)),
+                )),
+            )
+        };
+
+        let mut input = self.input.take().unwrap().execute();
+
+        // Consume the first barrier message and initialize state table.
+        let barrier = expect_first_barrier(&mut input).await?;
+        self.state_table.init_epoch(barrier.epoch);
+        let mut time_millis = Epoch::from(barrier.epoch.curr).as_unix_millis();
+        let mut sync_expr = None;
+
+        // The first barrier message should be propagated.
+        yield Message::Barrier(barrier);
+
+        // Fill the buffer on initialization. If the executor has recovered from a failed state, the
+        // buffer should be refilled by its previous data.
+        let vnode_bitmap = self.state_table.vnode_bitmap().to_owned();
+        self.fill_buffer(None, &vnode_bitmap).await?;
+
+        #[for_await]
+        for msg in input {
+            match msg? {
+                Message::Watermark(_) => {
+                    todo!("https://github.com/risingwavelabs/risingwave/issues/6042")
+                }
+                Message::Chunk(chunk) => {
+                    if sync_expr.is_none() {
+                        sync_expr = Some(dynamic_cond(time_millis)?);
+                        for stream_chunk in self.sync(sync_expr.as_ref().unwrap())? {
+                            yield Message::Chunk(stream_chunk);
+                        }
+                    }
+
+                    let (data_chunk, ops) = chunk.into_parts();
+                    let (new_ops, new_visibility) =
+                        self.apply_batch(&data_chunk, ops, sync_expr.as_ref().unwrap());
+                    if new_visibility.num_high_bits() > 0 {
+                        let (columns, _) = data_chunk.into_parts();
+                        let new_chunk = StreamChunk::new(new_ops, columns, Some(new_visibility));
+                        yield Message::Chunk(new_chunk);
+                    }
+                }
+                Message::Barrier(barrier) => {
+                    time_millis = Epoch::from(barrier.epoch.curr).as_unix_millis();
+                    if barrier.is_update() {
+                        sync_expr = None;
+                        self.state_table.commit_no_data_expected(barrier.epoch);
+                    } else {
+                        sync_expr = Some(dynamic_cond(time_millis)?);
+                        for stream_chunk in self.sync(sync_expr.as_ref().unwrap())? {
+                            yield Message::Chunk(stream_chunk);
+                        }
+                        self.state_table.commit(barrier.epoch).await?;
+                    }
+                    // Update the vnode bitmap for the state table if asked. Also update the buffer.
```

Comment:
Seems like this step only exists for `is_update()` branch? Can we put it in that branch?

Reply:
> Seems like this step only exists for `is_update()` branch? Can we put it in that branch?

Sorry. I just added `barrier.is_resume()` into the condition of the `is_update()` branch. Thank you.